caddyhttp/staticfiles/fileserver.go

```diff
@@ -40,6 +40,15 @@ func (fs FileServer) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, err
 	return fs.serveFile(w, r, r.URL.Path)
 }
 
+func isNotDirErr(err error) bool {
+	perr, ok := err.(*os.PathError)
+	if !ok {
+		return false
+	}
+
+	return perr.Err.Error() == "not a directory"
+}
+
```

Comment:
Yeah, I'm actually going to revisit this and update the PR with the fix I'm using/submitting to golang. I'll add the TODO and comments as well -- hopefully sometime today.